src/Service/OrderTotal/OrderTotalRestrictionService.php

```diff
@@ -0,0 +1,111 @@
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ *
+ * @author     Mollie B.V. <info@mollie.nl>
+ * @copyright  Mollie B.V.
+ * @license    Berkeley Software Distribution License (BSD-License 2) http://www.opensource.org/licenses/bsd-license.php
+ *
+ * @category   Mollie
+ *
+ * @see       https://www.mollie.nl
+ * @codingStandardsIgnoreStart
+ */
+
+namespace Mollie\Service\OrderTotal;
+
+use Currency;
+use Db;
+use Mollie\Exception\OrderTotalRestrictionException;
+use Mollie\Repository\CurrencyRepositoryInterface;
+use Mollie\Repository\PaymentMethodRepositoryInterface;
+use Mollie\Service\PaymentMethod\PaymentMethodOrderRestrictionUpdaterInterface;
+use MolPaymentMethod;
+use MolPaymentMethodOrderTotalRestriction;
+use PrestaShopCollection;
+
+class OrderTotalRestrictionService implements OrderTotalRestrictionServiceInterface
+{
+	/**
+	 * @var PaymentMethodRepositoryInterface
+	 */
+	private $paymentMethodRepository;
+
+	/**
+	 * @var CurrencyRepositoryInterface
+	 */
+	private $currencyRepository;
+
+	/**
+	 * @var PaymentMethodOrderRestrictionUpdaterInterface
+	 */
+	private $paymentMethodOrderRestrictionUpdater;
+
+	public function __construct(
+		PaymentMethodRepositoryInterface $paymentMethodRepository,
+		CurrencyRepositoryInterface $currencyRepository,
+		PaymentMethodOrderRestrictionUpdaterInterface $paymentMethodOrderRestrictionUpdater
+	) {
+		$this->paymentMethodRepository = $paymentMethodRepository;
+		$this->currencyRepository = $currencyRepository;
+		$this->paymentMethodOrderRestrictionUpdater = $paymentMethodOrderRestrictionUpdater;
+	}
+
+	/**
+	 * @throws OrderTotalRestrictionException
+	 */
+	public function updateOrderTotalRestrictions()
+	{
+		/** @var PrestaShopCollection $paymentMethods */
+		$paymentMethods = $this->paymentMethodRepository->findAll();
+
+		if (!$paymentMethods->count()) {
+			throw new OrderTotalRestrictionException('Failed to refresh order total restriction values: None available payment methods were found', OrderTotalRestrictionException::NO_AVAILABLE_PAYMENT_METHODS_FOUND);
+		}
+
+		/** @var PrestaShopCollection $currencies */
+		$currencies = $this->currencyRepository->findAll();
+
+		if (!$currencies->count()) {
+			throw new OrderTotalRestrictionException('Failed to refresh order total restriction values: None available currencies were found', OrderTotalRestrictionException::NO_AVAILABLE_CURRENCIES_FOUND);
+		}
+
+		/** @var Currency $currency */
+		foreach ($currencies as $currency) {
+			/** @var MolPaymentMethod $paymentMethod */
+			foreach ($paymentMethods as $paymentMethod) {
+				$this->paymentMethodOrderRestrictionUpdater->updatePaymentMethodOrderTotalRestriction(
+					$paymentMethod,
+					$currency->iso_code
+				);
+			}
+		}
+	}
```

Comment:
can we unit test this one ? You can mock stuff and for instance, expect that exception will be thrown or expect that some method will be called even with specific arguments